docs/tutorial/avax-nft.md

```diff
@@ -0,0 +1,530 @@
+
+async function storeNFT(imagePath, name, description) {
+    const image = await fileFromPath(imagePath)
+
+    const nftstorage = new NFTStorage({ token: NFT_STORAGE_KEY })
+
+    return nftstorage.store({
+        image,
+        name,
+        description,
+    })
+}
+
+async function fileFromPath(filePath) {
+    const content = await fs.promises.readFile(filePath)
+    const type = mime.getType(filePath)
+    return new File([content], path.basename(filePath), { type })
+}
+
+async function upload(imagePath, name, description) {
+    const result = await storeNFT(imagePath, name, description)
+    return result
+}
+
+export { upload }
+```
+
+Return to the Node REPL and import the `upload` function from the script. Copy an image file of your choice into the root directory to use as the NFT image.
+
+```js
+>> const { upload } = await import("./scripts/upload.mjs")
+>> const result = await upload("./pickleheart.png", "Pickleheart", "Image of Pickleheart Filet")
+>> result
+> Token {
+>   ipnft: 'bafyreicb3ewk33keh77mwxhmhdafxsjlkflichr2mjnyim6tbq3qjkwcue',
+>   url: 'ipfs://bafyreicb3ewk33keh77mwxhmhdafxsjlkflichr2mjnyim6tbq3qjkwcue/metadata.json'
+> }
+```
+
+Note we've received the response as a `Token` object with the `url` property. We can use this given URI to mint our first NFT to another account:
+
+```js
+>> const tokenId = await filet.callStatic.mintTo(accounts[1], result.url)
+>> tokenId.toNumber()
+> 1
+```
+
+Then check the balance of the receiving account:
+
+```js
+>> const balance = (await filet.balanceOf(account[1])).toString()
+>> balance
+> 1
+```
+
+Additionally, we can check the current owner of the NFT by calling `ownerOf(tokenId)`:
+
+```js
+>> const ownerAddress = (await filet.ownerOf(tokenId))
+>> ownerAddress === account[1].address
+> true
+```
+
+The receiving address now owns 1 FILET. We also confirmed that it is the owner of the first NFT.
+
+By default `ethers` uses the first address as the signer of the transaction, therefore it is `account[0]` signing off on the minting when we call `mintTo(recipientAddress, tokenURI)`.
+
+>> 💡 **Why did we call `callStatic`?**
+>> The return value of a non-pure or -view Solidity function is available only when the function is called on-chain (from the same contract or another contract).  
+>> When such function is called off-chain (i.e. from ethers.js as we are), the return value is the hash of that transaction. If we called `mintTo` directly, we would receive in return the transaction object, not the token ID we expect from the actual `mintTo` method in the contract.  
+>> To learn more, read [View and Pure Functions](https://solidity-by-example.org/view-and-pure-functions/), this [Stack Exchange post](https://ethereum.stackexchange.com/questions/88119/i-see-no-way-to-obtain-the-return-value-of-a-non-view-function-ethers-js), and [ethers.js doc for `callStatic`](https://docs.ethers.io/v5/single-page/#/v5/api/contract/contract/-%23-contract-callStatic).
```

Comment:
TIL about `callStatic` 😄 

The docs say:

> Rather than executing the state-change of a transaction, it is possible to ask a node to pretend that a call is not state-changing and return the result.

> This does not actually change any state, but is free. This in some cases can be used to determine if a transaction will fail or succeed.

From reading that, I would expect that you'd get a token id return value but that the state wouldn't actually change. But it seems to change the state in the example (changing the `ownerOf` account) ... maybe it works differently in the simulator? If so we should call that out.

Reply:
Either we should:
- Remove the use of `callStatic` and just add the part of handling this with event, which will add more complexity but sound reasonable and give the right way to do it
- Remove the use of `callStatic` and ignore the return value of non-view functions, and just assume the value (i.e. first token id must be tokenId = 1). This kind of glaze over the detail but keep the tutorial lean

Reply:
Decided to go with the shorter version, but expanded on this in the quote section followed.